File: familyPlanner/src/models/plannedTask.ts
import { Activity, ActivityInput } from "./activity";
import { Family } from "./family";
import { Meal, MealInput } from "./meal";
import { PlannedDay } from "./plannedDay";
import { ShoppingList, ShoppingListInput } from "./shoppingList";
import { TaskItem, TaskItemInput } from "./task";
import { User } from "./user";

export interface PlannedTask {
	id?: number;
	name: string;
	description?: string;
	status?: string;
	assignedTo?: User;
	startDate: Date;
	endDate: Date;
	type?: string;
	plannedDayId?: number;
	PlannedDay?: PlannedDay;
	familyId?: number;
	family?: Family;
	completed?: boolean;
}

export interface PlannedTaskInput {
	name: string;
	description?: string;
	status?: string;
	assignedTo?: User;
	startDate: Date;
	endDate: Date;
	type?: string;
	plannedDayId?: number;
	familyId?: number;
}

export interface PlannedTaskUpdate {
	name: string;
	description?: string;
	status?: string;
	assignedTo?: User;
	startDate: Date;
	endDate: Date;
	type?: string;
}

File: backend/models/PlannedTaskModel/PlannedTask.cs

using FamilyPlanner.Models.ActivityModel;
using FamilyPlanner.Models.FamilyModel;
using FamilyPlanner.Models.MealModel;
using FamilyPlanner.Models.PlannedDayModel;
using FamilyPlanner.Models.ShoppingListModel;
using FamilyPlanner.Models.TaskModel;
using FamilyPlanner.Models.UserModel;

namespace FamilyPlanner.Models.PlannedTaskModel {
    public class PlannedTask {
        public int Id { get; set; }
        public string Name { get; set; }
        public string? Description { get; set; }
        public string? Status { get; set; }
        public bool Completed { get; set; }
        public User? AssignedTo { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public string? Type { get; set; }
        public string? UserUid { get; set; }
        public int? PlannedDayId { get; set; }
        public PlannedDay? PlannedDay { get; set; }
        public int? FamilyId { get; set; }
        public Family? Family { get; set; }

        public override string ToString()
        {
            return $"id: {Id}, name: {Name}, description: {Description}";
        }
    }
}